winrm/client.go

```diff
@@ -107,49 +114,84 @@ func (client *Client) NewShell(shellID string) *Shell {
 }
 
 func (client *Client) sendRequest(request *soap.SoapMessage) (response string, err error) {
+	if errorCode, err := client.check(); errorCode != 0 {
+		return "", err
+	}
 	return client.http(client, request)
 }
 
+// Use this function to map any know terminating errors to exitcodes and messages
+func (client *Client) check() (exitCode int, err error) {
+	return FindErrorFromServerError(client.err)
+}
+
+// Check if we need to return an ExitCode
+func (client *Client) ExitCode(cmd *Command) (exitCode int) {
+	if cmd != nil {
+		if cmd.ExitCode() != 0 {
+			return cmd.exitCode
+		}
+	}
+	exitCode, _ = client.check()
+	return exitCode
```

Comment:
@hoenirvili ok, good to know.

We can make those functions not using named return variables, if that's the better way (I'm not an experienced Go dev as you might have noticed).

Regarding documenting our guide style, I'm not against it, but my experience shows that trying to adhere to very strict guidelines really reduces the number of people that are willing to help on a project. And I don't have the bandwidth to modify every pull request to adhere to a specific guide line.

Would there be a golang linter we can use that would check those kind of things?

Meanwhile I will modify this patch to eleminate named return vars where it's not needed.